### PR TITLE
Ruff set python version and enable more rules

### PR DESCRIPTION
**Pull request by tdadela**
1. Set python target-version for ruff.
2. Enable FA102 rule (Checks for uses of PEP 585- and PEP 604-style type annotations in Python modules that lack the required from __future__ import annotations import for compatibility with older Python versions).
3. Enable pyupgrade rules.
4. Make code compatible with rules.

### CODE REVIEW

File: locust/user/task.py [hunk 1]
```diff
@@ -423,10 +423,8 @@ class Tasks(TaskSet):
             return random.randint(self.min_wait, self.max_wait) / 1000.0
         else:
             raise MissingWaitTimeError(
-                "You must define a wait_time method on either the %s or %s class"
-                % (
-                    type(self.user).__name__,
-                    type(self).__name__,
+                "You must define a wait_time method on either the {} or {} class".format(
```
Comment (cyberw) [inline locust/user/task.py:426]: Maybe use regular f-string here too?

**Comment by cyberw** [comment]
Nice! One small comment, then it is good to go.

File: locust/user/task.py (hunk 1 above)
Comment (tdadela) [reply locust/user/task.py]: Done

**Comment by tdadela** [comment]
What do you think about also enabling isort-like "unsorted-imports (I001)" rule?
(https://docs.astral.sh/ruff/rules/unsorted-imports/)

**Comment by cyberw** [comment]
> What do you think about also enabling isort-like "unsorted-imports (I001)" rule? (https://docs.astral.sh/ruff/rules/unsorted-imports/)

Sounds good! I tried using isort a little, but in some cases (locustfiles) we need to ensure `locust` is imported before other things (because of gevent monkeypatching). If that can be solved then great!

**Pull request merged**